app/gui/src/project-view/composables/selection.ts

```diff
@@ -237,6 +246,10 @@ function useSelectionImpl<T, PackedT>(
         if (packed) rawSelected.add(packed)
       }
     },
+    setSoleSelected: (element: T) => {
+      setSelection(new Set([element]))
+      onSoleSelected?.(element)
+    },
```

Comment:
I think these names are a bit confusing. `setSoleSelected` technically should be the same as `setSelection` with a single node. But only the former calls `onSoleSelected`.